tools/src/format.rs

```diff
@@ -77,6 +77,15 @@ pub fn format_code(
 
     let mut cur_datum = 0;
 
+    while cur_datum < analysis.len() {
```

Comment:
It would be nice to have a comment here to provide context to someone newer to the codebase.  And/or if you want to extract `loc.lineno == 1 && loc.col_start == 0 && loc.col_end == 0` into a self-descriptive helper, that might make it more obvious what's going on.